makefiles/arch/atmega.inc.mk

```diff
@@ -0,0 +1,39 @@
+# Target architecture for the build. Use avr if you are unsure.
+TARGET_ARCH ?= avr
+
+CFLAGS_CPU   = -mmcu=$(CPU) $(CFLAGS_FPU)
+CFLAGS_LINK  = -ffunction-sections -fdata-sections -fno-builtin -fshort-enums
+CFLAGS_DBG  ?= -ggdb -g3
+CFLAGS_OPT  ?= -Os
+
+CFLAGS    += $(CFLAGS_CPU) $(CFLAGS_LINK) $(CFLAGS_DBG) $(CFLAGS_OPT)
+ASFLAGS   += $(CFLAGS_CPU) $(CFLAGS_DBG)
+LINKFLAGS += $(CFLAGS_CPU) $(CFLAGS_DBG) $(CFLAGS_OPT) -static -lgcc -e reset_handler -Wl,--gc-sections
+OFLAGS    += -j .text -j .data
+
+# Tell the build system that the CPU depends on the atmega common files:
+USEMODULE += atmega_common
+
+# export the peripheral drivers to be linked into the final binary
+USEMODULE += atmega_common_periph
+USEMODULE += periph_common
+
+# Export the peripheral drivers to be linked into the final binary, for now
+# only atmega126rfr2 has periph drivers
+ifeq ($(CPU), atmega256rfr2)
+  USEMODULE += periph
+endif
+
+# the atmel port uses uart_stdio
+USEMODULE += uart_stdio
+
+# explicitly tell the linker to link the syscalls and startup code.
+# without this the interrupt vectors will not be linked correctly!
+UNDEF += $(BINDIR)/atmega_common/startup.o
```

Comment:
Which is strange... shouldn't this be the same on all platforms? I tried both cases and both works... I'm missing something somehow.

Reply:
As far as I can track, this was introduced by the very first atmega platform supported (atmega2560) bye @N8Fear. Do you know why it was necessary (besides the comment about the vectors)? I'll make a test without and give feedback.

Reply:
Anything new on this front? It would be nice to drop the line (if no longer needed) or to extend the comment to be more elaborate on the technical details.